docs/releases/hotfix-release.md

```diff
@@ -128,8 +128,6 @@ As noted above, the challenges in creating a hotfix are unique to each attempt.
 
 1. Post-release steps
 
-   1. Delete the release branch.
```

Comment:
Let's leave it as is because sometimes release branches (including hotfix ones) aren't deleted